tests/test_premium.py

```diff
@@ -0,0 +1,422 @@
+
+        create_subscription(user, tier)
+
+        assert mock_stripe.Subscription.create.called
+        assert user.stripe_subscription_id is not None
+
+        # Check that the subscription ID was saved to the database
+        db.session.refresh(user)
+        assert user.stripe_subscription_id is not None
+
+
+def test_get_latest_invoice_payment_intent_client_secret(mock_stripe: MagicMock) -> None:
+    mock_stripe.Invoice.retrieve.return_value = MagicMock(payment_intent="pi_123")
+    mock_stripe.PaymentIntent.retrieve.return_value = MagicMock(client_secret="secret_123")  # noqa: S106
+
+    secret = get_latest_invoice_payment_intent_client_secret(MagicMock(latest_invoice="inv_123"))
+
+    assert secret == "secret_123"
+
+
+def test_get_subscription(app: Flask, mock_stripe: MagicMock) -> None:
+    mock_stripe.Subscription.retrieve.return_value = MagicMock(id="sub_123")
+
+    with app.app_context():
+        user = User(email="test@example.com", password="password")  # noqa: S106
+        user.stripe_subscription_id = "sub_123"
+        db.session.add(user)
+        db.session.commit()
+
+        subscription = get_subscription(user)
+
+        assert subscription is not None
+        assert mock_stripe.Subscription.retrieve.called
+
+
+def test_handle_subscription_created(app: Flask) -> None:
+    with app.app_context():
+        user = User(email="test@example.com", password="password")  # noqa: S106
+        user.stripe_customer_id = "cus_123"
+        db.session.add(user)
+        db.session.commit()
+
+        subscription = MagicMock()
+        subscription.customer = "cus_123"
+        subscription.id = "sub_123"
+        subscription.status = StripeSubscriptionStatusEnum.INCOMPLETE
+        subscription.cancel_at_period_end = False
+        subscription.current_period_end = datetime.now() + timedelta(days=30)
+        subscription.current_period_start = datetime.now()
+
+        handle_subscription_created(subscription)
+
+        assert user.stripe_subscription_id == "sub_123"
+        assert user.tier_id != BUSINESS_TIER
+
+
+def test_handle_subscription_updated_upgrade(app: Flask) -> None:
+    with app.app_context():
+        user = User(email="test@example.com", password="password")  # noqa: S106
+        user.stripe_subscription_id = "sub_123"
+        db.session.add(user)
+        db.session.commit()
+
+        subscription = MagicMock()
+        subscription.id = "sub_123"
+        subscription.status = StripeSubscriptionStatusEnum.ACTIVE
+        subscription.cancel_at_period_end = False
+        subscription.current_period_end = datetime.now() + timedelta(days=30)
+        subscription.current_period_start = datetime.now()
+
+        handle_subscription_updated(subscription)
+
+        assert user.tier_id == BUSINESS_TIER
+
+
+def test_handle_subscription_updated_downgrade(app: Flask) -> None:
+    with app.app_context():
+        user = User(email="test@example.com", password="password")  # noqa: S106
```

Comment:
`user` fixture

Reply:
Fixed in 180d5b3514b575c43e6b99320d8471a6f341796f